File: wlr/OGUR/OGUR/GameObjects/Altar.cs
using System.Collections.Generic;
using Microsoft.Xna.Framework;
using OGUR.Collision;
using OGUR.Items;
using OGUR.Sprites;
using OGUR.Management;
using OGUR.Dungeons;
using OGUR.Creatures;
using OGUR.Text;
using OGUR.Gods;
using System.Linq;

namespace OGUR.GameObjects
{
    public class Altar : Entity
    {
        private readonly God m_god;
        private Player m_currentTarget;
        private IEnumerable<Entity> m_offerings;

        public Altar(Point2 location,int godName)
        {
            m_god = God.Get(godName);
            m_graphic.SetColor(m_god.GetColor());
            Initialize(location, SpriteType.ALTAR, GameObjectType.ALTAR,Depth.Altar);
        }

        public override void Update()
        {
            m_currentTarget = EntityManager.GetTouchingPlayer(this);
            if (m_currentTarget != null)
            {
                if (m_currentTarget.IsInteracting())
                {
                    m_currentTarget.Pray(m_god);
                }
                m_offerings = EntityManager.GetObjects(GameObjectType.ITEM, m_location);
                foreach (GenericItem offering in m_offerings)
                {
                    m_currentTarget.Sacrifice(m_god, offering);
                }
                TextManager.Add(new ActionText(m_god.NameText, 1, (int) GetLocation().PosX, (int) GetLocation().PosY));
            }
        }

        public God GetGod()
        {
            return m_god;
        }
    }
}

File: proj/src/aigilas/entities/Altar.java
package aigilas.entities;

import aigilas.creatures.Player;
import aigilas.gods.God;
import aigilas.gods.GodId;
import aigilas.items.GenericItem;
import aigilas.management.SpriteType;
import spx.bridge.DrawDepth;
import spx.bridge.EntityType;
import spx.core.Point2;
import spx.entities.Entity;
import spx.entities.EntityManager;
import spx.entities.IEntity;
import spx.text.ActionText;
import spx.text.TextManager;

import java.util.List;

public class Altar extends Entity {
    private final God _god;
    private Player _currentTarget;
    private List<IEntity> _offerings;

    public Altar(Point2 location, GodId godName) {
        _god = godName.getInstance();
        _graphic.setColor(_god.getColor());
        initialize(location, SpriteType.ALTAR, EntityType.ALTAR, DrawDepth.Altar);
    }

    @Override
    public void update() {
        _currentTarget = (Player) EntityManager.getTouchingCreature(this);
        if (_currentTarget != null) {
            if (_currentTarget.isInteracting()) {
                _currentTarget.pray(_god);
            }
            _offerings = EntityManager.getEntities(EntityType.ITEM, _location);
            for (IEntity offering : _offerings) {
                _currentTarget.sacrifice(_god, (GenericItem) offering);
            }
            TextManager.add(new ActionText(_god.NameText, 1, (int) getLocation().PosX, (int) getLocation().PosY));
        }
    }

    public God getGod() {
        return _god;
    }
}
